src/edu/ucsb/cs56/projects/games/hangman/HangmanGUI.java

```diff
@@ -1,908 +1,822 @@
-	    // subtract from hints left (and one point) for each hint used unless zero hints remain
-	    if (hintsL > 0 && !hg.hasWon() && !hg.hasLost()  ) {
-		hintsL--;
-		if (numPoints > 0)
-		    numPoints--;
-		else
-		    numPoints = 0;
-		String numOfPoints = "Points: " + numPoints;
-		points.setText(numOfPoints);
-		char displayLetter = hg.getSecretWord().charAt(randomNum);
-		String alreadyGuessed = hg.getBoard();
-		while(alreadyGuessed.indexOf(displayLetter)!= -1){
-			randomNum = (int) (Math.random()*(wordLength));
-			displayLetter = hg.getSecretWord().charAt(randomNum);
-		    }
-		JOptionPane.showMessageDialog(f,"Your hint is:  "+displayLetter);
-		String hintsLeftString = "Hints left: " + hintsL;
-		hintsLeft.setText(hintsLeftString);
-	    }
-	    else
-		JOptionPane.showMessageDialog(f,"No hints remaining.");
-	}
+        public void actionPerformed(ActionEvent e) {
+            int wordLength = hg.getSecretWord().length();
+            int randomNum = (int) (Math.random() * (wordLength));
+
+            // subtract from hints left (and one point) for each hint used unless zero hints remain
+            if (hintsL > 0 && !hg.hasWon() && !hg.hasLost()) {
+                hintsL--;
+                if (numPoints > 0)
+                    numPoints--;
+                else
+                    numPoints = 0;
+                String numOfPoints = "Points: " + numPoints;
+                points.setText(numOfPoints);
+                char displayLetter = hg.getSecretWord().charAt(randomNum);
+                String alreadyGuessed = hg.getBoard();
+                while (alreadyGuessed.indexOf(displayLetter) != -1) {
+                    randomNum = (int) (Math.random() * (wordLength));
+                    displayLetter = hg.getSecretWord().charAt(randomNum);
+                }
+                JOptionPane.showMessageDialog(f, "Your hint is:  " + displayLetter);
+                String hintsLeftString = "Hints left: " + hintsL;
+                hintsLeft.setText(hintsLeftString);
+            } else
+                JOptionPane.showMessageDialog(f, "No hints remaining.");
+        }
     }
 
-   
-    /** Handler for options button
-	Button displays new JFrame with option selections to choose
-    */
-    public class optionsButtonHandler implements ActionListener {
-	public void actionPerformed(ActionEvent e) {
-        f.setEnabled(false);
-	    buildOptions();
-	}
-    }
-    
 
-    /** Handler for finish button
+    /**
+     * Handler for options button
+     * Button displays new JFrame with option selections to choose
      */
-    public class finishButtonHandler implements ActionListener {
-	public void actionPerformed(ActionEvent e) {
-	    optionsOn = "true"; // intended to save and active chosen options
-        f.setEnabled(true);
-	    o.dispose();
-	}
+    public class optionsButtonHandler implements ActionListener {
+        HangmanGUI gui;
+        public optionsButtonHandler(HangmanGUI gui){
+            this.gui = gui;
```

Comment:
FYI... https://stackoverflow.com/questions/31201/how-do-you-get-a-reference-to-the-enclosing-class-from-an-anonymous-inner-class